tests/integration/test_studio_editable.py

```diff
@@ -0,0 +1,299 @@
+        if data.count <=0:
+            validation.add(ValidationMessage(ValidationMessage.ERROR, u"Count cannot be negative"))
+        if "damn" in data.comment.lower():
+            validation.add(ValidationMessage(ValidationMessage.ERROR, u"No swearing allowed"))
+
+
+class TestEditableXBlock_StudioView(StudioEditableBaseTest):
+    """
+    Test the Studio View created for EditableXBlock
+    """
+    @XBlock.register_temp_plugin(EditableXBlock, "editable")
+    def setUp(self):
+        super(TestEditableXBlock_StudioView, self).setUp()
+        self.set_scenario_xml('<editable />')
+        self.go_to_view("studio_view")
+        self.fix_js_environment()
+
+    def test_no_changes_with_defaults(self):
+        """
+        If we load the edit form and then save right away, there should be no changes.
+        """
+        block = self.load_root_xblock()
+        orig_values = {field_name: getattr(block, field_name) for field_name in EditableXBlock.editable_fields}
+        self.click_save()
+        for field_name in EditableXBlock.editable_fields:
+            self.assertEqual(getattr(block, field_name), orig_values[field_name])
+            self.assertFalse(block.fields[field_name].is_set_on(block))
+
+    def test_no_changes_with_values_set(self):
+        """
+        If the XBlock already has explicit values set, and we load the edit form and then save
+        right away, there should be no changes.
+        """
+        block = self.load_root_xblock()
+        block.color = "green"
+        block.count = 5
+        block.comment = "Hello"
+        block.save()
+        orig_values = {field_name: getattr(block, field_name) for field_name in EditableXBlock.editable_fields}
+        # Reload the page:
+        self.go_to_view("studio_view")
+        self.fix_js_environment()
+
+        self.click_save()
+        block = self.load_root_xblock()  # Need to reload the block to bypass its cache
+        for field_name in EditableXBlock.editable_fields:
+            self.assertEqual(getattr(block, field_name), orig_values[field_name])
+            self.assertTrue(block.fields[field_name].is_set_on(block))
+
+    def test_explicit_overrides(self):
+        """
+        Test that we can override the defaults with the same value as the default, and that the
+        value will be saved explicitly.
+        """
+        block = self.load_root_xblock()
+        for field_name in EditableXBlock.editable_fields:
+            self.assertFalse(block.fields[field_name].is_set_on(block))
+
+        color_control = self.get_element_for_field('color')
+        color_control.clear()
+        color_control.send_keys('red')
+
+        count_control = self.get_element_for_field('count')
+        count_control.clear()
+        count_control.send_keys('42')
+
+        comment_control = self.get_element_for_field('comment')
+        comment_control.send_keys('forcing a change')
+        comment_control.clear()
```

Comment:
Why do you call the clear() after the send_keys() here, but before it in the previous two stanza?

I also wonder if there's a way to do these three (get element, clear, send_keys) in a loop over EditableXBlock.editable_fields (maybe you can send the default value or something?) so that this test is more robust to adding of any new editable fields.